use setting for case search index name

snopoke opened the pull request:
## Summary
Get the name of the case search index from settings to allow overriding it in localsettings.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

snopoke committed:
make case search index name configurable

dannyroberts commented on corehq/ex-submodules/pillowtop/es_utils.py:
```diff
@@ -79,6 +79,13 @@ def _get_analysis(*names):
+                "number_of_shards": 50,
+            },
+        },
+    }
```
Could this be done in commcare-cloud `production/elasticsearch.yml` instead? Or is there something that can be only done here?

See https://github.com/dimagi/commcare-cloud/blob/da13063249289b7208bb4e1a7bd8bde1362e618f/environments/staging/elasticsearch.yml#L3 as an example of this file. It's used to populate the `ES_SETTINGS` variable in `localsettings.py`

snopoke replied:
It can be done in commcare-cloud but the process is bit trickier since we want this present in the limited release but not in the current release. Perhaps it doesn't matter since I don't think it get's used outside of creating the index and perhaps a few management commands.

If this were tied to the actual index name I would feel a bit safer since then it would only apply to the new index and not the old one.

dannyroberts commented on corehq/pillows/mappings/case_search_mapping.py:
```diff
@@ -4,8 +4,8 @@
-CASE_SEARCH_INDEX = prefix_for_tests("case_search_2018-05-29")
-CASE_SEARCH_ALIAS = prefix_for_tests("case_search")
+CASE_SEARCH_INDEX = prefix_for_tests("case_search_2021-06-01")
+CASE_SEARCH_ALIAS = prefix_for_tests("case_search-1")
```
It looks like this (trivally) breaks the build. Is the idea here to make a dummy alias that we then switch over to? Is there any reason we can't keep the original alias name and just not flip the alias until the end? I feel like there's something you thought of here that I'm missing

snopoke replied:
The way the code is currently written we assign the alias when the index is created. I think we can revert the alias change once we're ready to roll out. Alternatively I would need need to modify the code to make assigning the alias optional.

sravfeyn commented:
What's the plan to have other envs roll this out? Would that happen as part of deploy/preindex?

snopoke commented:
> What's the plan to have other envs roll this out? Would that happen as part of deploy/preindex?

I think we'll most likely make this env specific so that we don't need to make changes to other envs.

sravfeyn commented:
> I think we'll most likely make this env specific so that we don't need to make changes to other envs.

If we update the index names, when other envs deploy, that would trigger a reindex automatically, right? If the other envs are fairly small, it probably doesn't matter. But if any of their indices are large, they would also need to reindex using a private release like this (?). Though, may be there aren't any.

snopoke commented:
@dannyroberts @sravfeyn I've rebased this and removed any actual changes so this is now just moving the current index name to settings and is safe to merge.
